src/app/shared/services/ssh-keypair.service.ts

```diff
@@ -20,15 +21,23 @@ export interface SshKeyCreationData {
   entityModel: SSHKeyPair
 })
 export class SSHKeyPairService extends BaseBackendCachedService<SSHKeyPair> {
-  constructor(
-    private asyncJobService: AsyncJobService,
-    protected http: HttpClient
-  ) {
+  protected account: string;
+
+  constructor(private asyncJobService: AsyncJobService,
+              protected http: HttpClient,
+              protected activatedRoute: ActivatedRoute) {
     super(http);
+
+    this.activatedRoute.queryParams.subscribe((params: Params) => {
+      this.account = params['account'];
+    });
   }
 
   public getByName(name: string): Observable<SSHKeyPair> {
-    return this.getList({ name }).map(sshKeys => sshKeys[0]);
+    return this.getList({ name }).map(sshKeys => {
```

Comment:
is it possible to fetch only one key by account **and** name? there will be no need to do subscription in constructor and store it locally

Reply:
You also have to handle the situation when account is not present in the query string